content/handling-multiple-k8sclusters.md

```diff
@@ -0,0 +1,69 @@
+
+Once you have set up your clusters, users and contexts, you can easily switch between contexts using the `kubectl config use-context` command. The currently active context can be checked with `kubectl config current-context`. 
+
+A more detailed description of all this can be found at [https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/]. 
+
+Theoretically, this could be the end of the article, if it wasn't for the fact that once the number of clusters you interact with grows beyond low single-digit numbers, keeping all information in one config file becomes, let's say, unpleasant. 
+
+## Multiple config files
+
+A better way (in my humble opinion) is to keep information about different clusters in different config files. There's basically two ways to handle multiple config files: the simple way and the clever way. The simple way is you manually specify the config you want to use each time you call kubectl via the `--kubeconfig` flag. The clever way is to concatenate multiple kubeconfigs into your `KUBECONFIG` environment variable like 
+
+    export KUBECONFIG=$HOME/.kube/config:second-config-file:third-config-file. 
+
+On Linux and Mac, the list is delimited by a colon, while for Windows you'll use a semicolon. 
+
+The rules according to which kubectl identifies its valid configuration data from this list of files are described in detail in the [Kubernetes documentation](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/), but here's the 10000 m summary:
+* Files specified via --kubeconfig always win, no matter what's in your KUBECONFIG 
+* The content of all configs specified in $KUBECONFIG is merged. You can use `kubectl config view --merge` to check what this leads to on your system. 
+* The first file to set a value or map key wins
+* Never change a value or map key. Even if a later occurrence has non-conflicting values, discard that info
+* The first context found in the chain wins - if you have a context with the same name in a later file, kubectl will never find it. 
+
+In case you ended up with multiple configs in one file, and you want to separate them, you can extract all information for one specific context with
+
+    kubectl config view --minify --flatten --context=my_context > myconfig.yaml
+
+## Auto-populate $KUBECONFIG
+
+If the set of clusters you interact with changes often, one elegant way to deal with that is to populate your KUBECONFIG variable automatically. There's many ways to do this, but here's what I'm using. I have this line in my environment:
+
+    set -xg KUBECONFIG $HOME/.kube/config:(ls -m $HOME/.kube/*.yaml | tr ',' ':' | tr -d "[:blank:]") 
+
+ Note that I'm using fish as my default shell, hence I'm using fish's native `set` command with the `-x` flag instead of `export`. In a normal bash or zsh environment you would use `export KUBECONFIG=...` for the same purpose. 
```

Comment:
There might be room for a whole "how to use fish with kubernetes everything"

Reply:
Probably not. I wrote this before seeing your comments later on. One problem that I run into commonly is that people write installation instructions for their tools that assume bash and use substitutions in a way that I can't just copy and paste

Reply:
Which is out of scope for this article anyways